### ISSUE TITLE
Add resource dependency definition back to Spec

### ISSUE THREAD

**Issue by hongchaodeng** [oam-dev/spec#386]
[Dependency management feature](https://github.com/crossplane/oam-kubernetes-runtime/blob/master/design/resource-dependency.md) has been implementedin [oam-k8s-runtime](https://github.com/crossplane/oam-kubernetes-runtime/tree/master/examples/dependency). We should add it back into spec.

**Comment by hongchaodeng** [comment]
Another solution is that we add an `experimental` folder in spec to record latest features available in oam-k8s-runtime until they are made stable in the Spec.

**Comment by resouer** [comment]
Yes, let's start from a experimental dir in the spec repo like this: https://github.com/open-telemetry/opentelemetry-specification/tree/master/experimental
